static/app/utils/withRelease.spec.jsx

```diff
@@ -46,58 +46,57 @@ describe('withRelease HoC', function () {
       />
     );
 
-    await tick(); // Run Store.loadEtc
-    await tick(); // Run Store.loadEtcSuccess
-    wrapper.update();
-
-    const mountedComponent = wrapper.find(Component);
-    expect(mountedComponent.prop('release')).toEqual(mockData);
-    expect(mountedComponent.prop('releaseLoading')).toEqual(false);
-    expect(mountedComponent.prop('releaseError')).toEqual(undefined);
-    expect(mountedComponent.prop('deploys')).toEqual([mockData]);
-    expect(mountedComponent.prop('deploysLoading')).toEqual(false);
-    expect(mountedComponent.prop('deploysError')).toEqual(undefined);
+    await waitFor(() =>
+      expect(Component).toHaveBeenCalledWith(
+        expect.objectContaining({
+          release: mockData,
+          releaseLoading: false,
+          releaseError: undefined,
+          deploys: [mockData],
+          deploysLoading: false,
+          deploysError: undefined,
+        }),
+        {}
+      )
+    );
   });
 
-  it('prevents repeated calls', async () => {
-    const Component = () => null;
+  it('prevents repeated calls', function () {
+    const Component = jest.fn(() => null);
     const Container = withRelease(Component);
 
     jest.spyOn(api, 'requestPromise');
     jest.spyOn(Container.prototype, 'fetchRelease');
     jest.spyOn(Container.prototype, 'fetchDeploys');
 
     // Mount and run component
-    mountWithTheme(
+    render(
       <Container
         api={api}
         organization={organization}
         projectSlug={projectSlug}
         releaseVersion={releaseVersion}
       />
     );
-    await tick();
-    await tick();
 
     // Mount and run duplicates
-    mountWithTheme(
+    render(
```

Comment:
hm this looks good, but don't we want to use `rerender` here?

Reply:
Nah this is actually testing that rendering _multiple_ components with `withRelease` doesn't duplicate requests